### ISSUE TITLE
Custom adpotopenjdk url causes directory name parsing error

### ISSUE THREAD

**Issue by pavannaganna** [sous-chefs/java#557]
### Cookbook version
4.1.0

### Chef-client version
14.13.11

### Platform Details
RHEL7

### Scenario:
Install the flavor adoptopenjdk using custom url which is of type .tgz (but not tar.gz)

### Steps to Reproduce:
Use the remote URL having .tgz as file extension. This causes the regex having (.tar.gz) to fail. 

### Expected Result:
Install the adoptopenjdk JDK 

### Actual Result:
Error 
```Recipe: java::notify
  * log[jdk-version-changed] action nothing (skipped due to action :nothing)
Recipe: java::set_java_home
  * directory[/etc/profile.d] action create (up to date)
  * template[/etc/profile.d/jdk.sh] action create
    - update content in file /etc/profile.d/jdk.sh from 321cfd to 2fc8a4
    --- /etc/profile.d/jdk.sh   2019-07-01 18:39:09.050393000 +0000
    +++ /etc/profile.d/.chef-jdk20190701-1434-1xzao5f.sh        2019-07-01 18:52:25.427440000 +0000
    @@ -1,2 +1,2 @@
    -export JAVA_HOME=/usr/lib/jvm/java-11-adoptopenjdk-openj9
    +export JAVA_HOME=/usr/lib/jvm/java-11-adoptopenjdk-hotspot
Recipe: java::adoptopenjdk
  * adoptopenjdk_install[adoptopenjdk] action install[2019-07-01T18:52:25+00:00] FATAL: Failed to parse company_jdk-11.0.3_7.325.tgz for application directory name!


  Running handlers:
[2019-07-01T18:52:25+00:00] ERROR: Running exception handlers
  Running handlers complete
[2019-07-01T18:52:25+00:00] ERROR: Exception handlers complete
  Chef Client failed. 1 resources updated in 03 seconds
[2019-07-01T18:52:25+00:00] FATAL: Stacktrace dumped to /opt/kitchen/cache/chef-stacktrace.out
[2019-07-01T18:52:25+00:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report```

**Comment by damacus** [comment]
Fixed via #589

**Comment by damacus** [comment]
The custom URL will also break the new resource, as it helps discover the `java_home`. In that case, all you need to do is manually set `java_home`, or have the same subdirectory in the URL as the original URL
